src/js/background/background.js

```diff
@@ -166,149 +210,8 @@ async function makeRelayAddress(description = null) {
-  // Catch errors when trying to create the same menu twice.
-  // The browser.menus API is limited. You cannot query if a menu item already exists.
-  // The error it throws does not show up to the user. 
-  if (browser.runtime.lastError) {
-    return;
-  }
-}
-
-async function createMenu() {
-  if (browser.menus) {
-    browser.menus.create({
-      id: "fx-private-relay-generate-alias",
-      title: browser.i18n.getMessage("pageInputIconGenerateNewAlias"),
-      contexts: ["editable"],
-    }, onContextMenuCreated);
-  }
-}
-
-
-createMenu();
-
-async function updateGenerateAliasContextMenuItem(relayAddressesLength) {
-  const { premiumEnabled } = await browser.storage.local.get("premiumEnabled");
-  const { premium } = await browser.storage.local.get("premium");
-  const { maxNumAliases } = await browser.storage.local.get("maxNumAliases");
-  const aliasesRemaining = maxNumAliases - relayAddressesLength;
-  
-  if (premiumFeaturesAvailable(premiumEnabled)) {
-    if (!premium && aliasesRemaining < 1) {
-      // Post-launch: Check if user is premium and under the max limit.
-      browser.menus.remove("fx-private-relay-generate-alias");
-      return;
-    }
-  } else {
-    // TODO: REMOVE THIS BLOCK AFTER PREMIUM LAUNCH
-    if (aliasesRemaining < 1) {
-      // Current users
-      browser.menus.remove("fx-private-relay-generate-alias");
-      return;
-    }
-  }
-
-  // Generate Alias link should be visible
-  createMenu();
-}
-
-async function createUpgradeContextMenuItem() {
-  browser.menus.create({
-    id: "fx-private-relay-get-unlimited-aliases",
-    title: browser.i18n.getMessage("pageInputIconGetUnlimitedAliases"),
-  });
-}
-
-function removeUpgradeContextMenuItem() {
-  browser.menus.remove("fx-private-relay-get-unlimited-aliases");
-}
-
-async function updateUpgradeContextMenuItem() {
-  await refreshAccountPages();
-  // Check for status update
-  const { premiumEnabled } = await browser.storage.local.get("premiumEnabled");
-  const { premium } = await browser.storage.local.get("premium");
-
-  if (premiumFeaturesAvailable(premiumEnabled)) {
-    if (!premium) {
-      // Remove any previous upgrade menu items first!
-      removeUpgradeContextMenuItem();
-      await createUpgradeContextMenuItem();
-      return;
-    }
-
-    // Remove the upgrade item, if the user is upgraded
-    else {
-      removeUpgradeContextMenuItem();
-    }
-  }
-}
-
-async function refreshAccountPages() {
-  const { settingsRefresh } = await browser.storage.local.get("settingsRefresh");
-
-  // This functions only runs once (when on the dashboard page), if the user has visited the settings page.
-  // If they revisit the settings page, it resets so that it only runs once again. 
-  if (!settingsRefresh) {
-    browser.storage.local.set({ settingsRefresh: true });
-
-    browser.tabs.query({ url: "http://127.0.0.1/*" }, function (tabs) {
-      for (let tab of tabs) {
-        browser.tabs.reload(tab.id);
-      }
-    });
-  }
```

Comment:
Reviewer note: This code was moved to its own partial, and was then further refactored.